File: CPP/Assignment3/Program5.cpp
//////////////////////////////////////////////////////////////////////////////////////////
//
// File name :   Program5.cpp
// Description : Write generic program which accept N values & reverse the contents.
// Author :      Pallavi Omprakash Malewar
// Date :        27/07/2026
//
//////////////////////////////////////////////////////////////////////////////////////////

/*

    Input :     10  20  30  10  30  40  10  40  10

    Output :    10  40  10  40  30  10  30  20  10

*/


#include<iostream>
using namespace std;

template<class T>

void Reverse(T *arr, int iSize)
{
    int i = 0;
    T temp = T();
    for(i = 0 ; i < iSize/2 ; i++)
    {
        temp = *(arr + i);
        *(arr + i) = *(arr + iSize -1 -i);
        *(arr + iSize -1 -i) = temp;
    }
    for(i = 0 ; i < iSize ; i++)
    {
        cout << *(arr + i) << "\t";
    }
    cout << endl;
}

int main()
{
    int arr[] = {10, 20, 30, 10, 30, 40, 10, 40, 10};
    Reverse(arr, 9);

    return 0;
}

File: JAVA/Assignment 7/Program5.java
/////////////////////////////////////////////////////////////////////////////////////////
//                                                                                     //
// Problem Statement : Write Java program which accept two arrays from user & check    //
//                     whether that array & its elements are palindrome or not.        //
// Author : Pallavi Omprakash Malewar                                                  //
// Date : 30 Jun 2026                                                                  //
//                                                                                     //
/////////////////////////////////////////////////////////////////////////////////////////


/*  

    Input : 2   9   7   7   9   2

    Output: TRUE

    Input : 2   9   7   6   4   5

    Output: FALSE

*/


import java.util.Scanner;

class MyArray
{
    public boolean CopyArrayRev(int Arr[])
    {
        int n = Arr.length;

        int i = 0;
        for(i = 0 ; i < n-1 ; i++)
        {
            if(Arr[i] != Arr[n-i-1]) return false;
        }
        return true;
    }
}

class Program5
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.print("Enter the Size of Array : ");
        int No = sobj.nextInt();

        int Arr[] = new int[No];
        boolean bRet = false;

        for(int i = 0 ; i < No ; i++)
        {
            Arr[i] = sobj.nextInt();
        }

        MyArray mobj = new MyArray();

        bRet = mobj.CopyArrayRev(Arr);
        
        if(bRet == true) System.out.println("TRUE");
        else System.out.println("FALSE");
    }
}
